mlflow/projects/__init__.py

```diff
@@ -812,7 +813,11 @@ def _get_local_artifact_cmd_and_envs(artifact_repo):
 
 def _get_s3_artifact_cmd_and_envs(artifact_repo):
     # pylint: disable=unused-argument
-    aws_path = posixpath.expanduser("~/.aws")
+    if platform.system() == "Windows":
+        win_user_dir = os.environ["userprofile"]
```

Comment:
Use the [env variable in upper case](https://docs.microsoft.com/en-us/windows/deployment/usmt/usmt-recognized-environment-variables): `os.environ["USERPROFILE"]`. Windows OS variables are case insensitive, but traditionally written as upper case.